routes/index.js

```diff
@@ -150,6 +165,75 @@ module.exports = function(utils, env, nunjucksEnv, appName) {
       app.get( '/slowparse/spec/errors.forbidjs.html', function( req, res ) {
         res.render('/slowparse/spec/errors.forbidjs.html');
       });
+    },
+
+    oauth2Callback: function(req, res, next) {
+      if (req.query.logout) {
+        req.session = null;
+        return res.redirect(301, '/');
+      }
+
+      if (!req.query.code) {
```

Comment:
Not sure if we want to follow "best" (but annoying) practice here in not revealing *why* there's a 401, just that *something* was wrong leading to a 401